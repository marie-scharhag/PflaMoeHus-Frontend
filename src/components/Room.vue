<template>
  <div class="room">
    <img v-bind:src="require(`../assets/${link}`)" class="pic" alt="Picture" />
    <div class="roomName">{{ name }}</div>
  </div>
</template>

<script lang="ts">
import { defineComponent } from "vue";
export default defineComponent({
  name: "Room",
  props: {
    name: String,
    link: String,
  },
});
</script>

<style scoped lang="scss">
img {
  border-radius: 10px 10px 0px 0px;
  width: 400px;
  height: 300px;
  object-fit: cover;
  object-position: bottom center;
}

.roomName{
    font-size: 1.5em;
    padding-top: 0.7em;
    padding-bottom: 0.7em;
    background-color: $color-grey;
    width:400px;
    text-align: center;
    border-radius: 0px 0px 10px 10px;
    color: black;
}
.room {
  display: table;
  padding: 2em;
}
</style>