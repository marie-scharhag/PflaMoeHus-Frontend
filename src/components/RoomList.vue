<template>
  <div class="roomList">
    <router-link
      :to="{
        path: '/product',
        query: { room: 'BATHROOM', productType: 'all', name: 'none' },
      }"
      ><Room class="listOrder" name="Bad" link="bathroom.jpg"
    /></router-link>
    <router-link
      :to="{
        path: '/product',
        query: { room: 'BEDROOM', productType: 'all', name: 'none' },
      }"
      ><Room class="listOrder" name="Schlafzimmer" link="sleepingroom.png"
    /></router-link>
    <router-link
      :to="{
        path: '/product',
        query: { room: 'KITCHEN', productType: 'all', name: 'none' },
      }"
      ><Room class="listOrder" name="KÃ¼che" link="cookingroom.jpg"
    /></router-link>
    <router-link
      :to="{
        path: '/product',
        query: { room: 'LIVINGROOM', productType: 'all', name: 'none' },
      }"
      ><Room class="listOrder" name="Wohnzimmer" link="livingroom.jpg"
    /></router-link>
    <router-link
      :to="{
        path: '/product',
        query: { room: 'DININGROOM', productType: 'all', name: 'none' },
      }"
      ><Room class="listOrder" name="Esszimmer" link="eatingroom.jpg"
    /></router-link>
    <router-link
      :to="{
        path: '/product',
        query: { room: 'BUREAU', productType: 'all', name: 'none' },
      }"
      ><Room class="listOrder" name="Arbeitszimmer" link="workingroom.jpg"
    /></router-link>
  </div>
</template>

 <script>
import Room from "@/components/Room.vue";

export default {
  components: { Room },
};
</script>

<style scoped lang="scss">
.roomList{
    padding-left: 1.5em;
}

.listOrder {
  float: left;
}
</style>