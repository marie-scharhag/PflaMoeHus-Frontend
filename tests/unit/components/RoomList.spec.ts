import { mount, RouterLinkStub } from "@vue/test-utils"
	
import RoomList from "@/components/RoomList.vue"
import Room from "@/components/RoomList.vue"
	
describe('Test room.vue', () => {
    const cmp = mount(RoomList)
	
    test('has the expected html structur', () => {
        expect(cmp.html()).toMatchSnapshot()
    })
	
    test('props test, class contains exactly 2 props', () => {
        expect(cmp.findComponent(Room)).toBeTruthy()
    })

    test('if roomlist is complete(HTML)', () => {
        expect(cmp.html()).toContain('Bad')
        expect(cmp.html()).toContain('Schlafzimmer')
        expect(cmp.html()).toContain('KÃ¼che')
        expect(cmp.html()).toContain('Wohnzimmer')
        expect(cmp.html()).toContain('Esszimmer')
        expect(cmp.html()).toContain('Arbeitszimmer')
    })
})