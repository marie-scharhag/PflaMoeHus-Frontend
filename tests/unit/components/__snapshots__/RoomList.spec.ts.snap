// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test room.vue has the expected html structur 1`] = `
<div class="roomList">
  <router-link to="[object Object]">
    <div class="room listOrder"><img src="" class="pic" alt="Picture">
      <div class="roomName">Bad</div>
    </div>
  </router-link>
  <router-link to="[object Object]">
    <div class="room listOrder"><img src="" class="pic" alt="Picture">
      <div class="roomName">Schlafzimmer</div>
    </div>
  </router-link>
  <router-link to="[object Object]">
    <div class="room listOrder"><img src="" class="pic" alt="Picture">
      <div class="roomName">KÃ¼che</div>
    </div>
  </router-link>
  <router-link to="[object Object]">
    <div class="room listOrder"><img src="" class="pic" alt="Picture">
      <div class="roomName">Wohnzimmer</div>
    </div>
  </router-link>
  <router-link to="[object Object]">
    <div class="room listOrder"><img src="" class="pic" alt="Picture">
      <div class="roomName">Esszimmer</div>
    </div>
  </router-link>
  <router-link to="[object Object]">
    <div class="room listOrder"><img src="" class="pic" alt="Picture">
      <div class="roomName">Arbeitszimmer</div>
    </div>
  </router-link>
</div>
`;
